File: Hackerrank/30_days_of_code_csharp/day_17.cs
using System;

class Calculator {
    public int power(int n, int p) {
        if (n < 0 || p < 0) throw new Exception("n and p should be non-negative");
        else return (int)Math.Pow(n, p);
    }
}

class Solution {
    static void Main(String[] args) {
        Calculator myCalculator = new  Calculator();
        int T = Int32.Parse(Console.ReadLine());
        while (T-- > 0) {
            string[] num = Console.ReadLine().Split();
            int n = int.Parse(num[0]);
            int p = int.Parse(num[1]);
            try {
                Console.WriteLine(myCalculator.power(n,p));
            }
            catch (Exception e) {
               Console.WriteLine(e.Message);
            }
        }
    }
}

File: Hackerrank/30_days_of_code_cpp/day_17.cpp
#include <cmath>
#include <iostream>
#include <exception>
#include <stdexcept>

using namespace std;

class PowerException: public exception {
    virtual const char* what() const throw() {
        return "n and p should be non-negative";
    }
};

class Calculator {
    public:
        int power(int n, int p) {
            if(n < 0 || p < 0) {
                throw PowerException();
            }
            return int(pow(n, p));
        }
};

int main() {
    Calculator calc = Calculator();
    int T, n, p;
    cin >> T;
    while (T-->0) {
        if (scanf("%d %d", &n, &p) == 2) {
            try {
                cout << calc.power(n,p) << endl;
            }
            catch (exception &e) {
                cout << e.what() << endl;
            }
        }
    }
}
